Add support to automate setup and config of RDS & S3 for openshift-metering

chancez opened the pull request:
Add new tasks to the openshift_metering role for installing/uninstalling RDS & S3 for use with Metering, and update existing tasks to configure metering to use the configured RDS instance and/or S3 bucket.

Metering currently supports storing data in HDFS and S3, this just automates using S3.
The metadata is managed by Hive, which by default persists it's data in a local embedded database stored in a persistent volume, using RDS allows it to persist it's data in an external SQL database, and improves scalability over the embedded database.

sdodson commented:
We try not to use set_fact, I read through this any I think all uses of set_fact can just be variables defined in defaults/main.yml

sdodson commented:
/cc @michaelgugino

sdodson commented:
Travis errors.

```
roles/openshift_metering/tasks/common-facts.yaml:
  9:1       error    too many blank lines (1 > 0)  (empty-lines)
roles/openshift_metering/tasks/rds-database-uninstall.yaml:
  1:1       warning  missing document start "---"  (document-start)
```

chancez commented:
Looking at jenkins it seems like it's timing out on AWS provisioning.
```
TASK [aws-up : wait for SSH to be available] ***********************************
```

michaelgugino commented on roles/openshift_metering/tasks/main.yaml:
```diff
@@ -1,6 +1,30 @@
-- include_tasks: install.yaml
-  when: openshift_metering_install | default(true) | bool
+- name: Openshift Metering RDS Database setup
+  include_tasks: rds-database-install.yaml
```
import_tasks is preferred over include_tasks.

https://docs.ansible.com/ansible/2.5/user_guide/playbooks_reuse_includes.html

chancez replied:
It's okay to only do it for the ones which don't use loops right?

chancez commented:
@michaelgugino Updated to use imports where possible.

chancez commented:
Also just pushed up a squashed commit with all the changes included.

chancez commented:
I guess I need to figure out how to deal with this properly.

```
Failure summary:


  1. Hosts:    ci-op-7ivxs231-1827f-ig-m-cwh6
     Play:     Configure Openshift Metering
     Task:     Delete  Metering Hive metadata RDS database
     Message:  Region not specified. Unable to determine region from EC2_REGION.
2018/08/01 20:30:28 Container setup in pod e2e-gcp failed, exit code 2, reason Error
```

chancez commented:
Updated it so that uninstall of cloud resources isn't done by default, and variables need to be set to deprovision them. In the future, the playbook can expose different roles for install vs uninstall.

michaelgugino commented on roles/openshift_metering/tasks/main.yaml:
```diff
@@ -1,6 +1,30 @@
+- name: Openshift Metering RDS Database setup
+  import_tasks: rds-database-install.yaml
+  when: openshift_metering_hive_metastore_db_rds_create | bool
+  tags: [metering-hive-rds, metering-hive-rds-install]
```
No tags allowed.

chancez replied:
Done

michaelgugino commented on roles/openshift_metering/tasks/main.yaml:
```diff
@@ -1,6 +1,30 @@
+  tags: metering-install
+
+- name: Openshift Metering Operator teardown
+  import_tasks: uninstall.yaml
```
Move uninstall portions to their own play.

chancez replied:
Done

chancez commented:
Broke install/uninstall into their own plays. I left config play alone since I tend to prefer using a single playbook and modifying variables, but I think that should be okay, as the install/uninstall plays are available.

chancez commented:
Fixed the last failure. Renamed the private playbook, but the components playbook referenced the old name, so I just renamed it back to private/config.yml.
